src/openrct2/actions/ChangeMapSizeAction.cpp

```diff
@@ -33,11 +33,11 @@ void ChangeMapSizeAction::Serialise(DataSerialiser& stream)
 
 GameActions::Result ChangeMapSizeAction::Query() const
 {
-    if (_targetSize > MAXIMUM_MAP_SIZE_TECHNICAL)
+    if (_targetSize.x > MAXIMUM_MAP_SIZE_TECHNICAL || _targetSize.y > MAXIMUM_MAP_SIZE_TECHNICAL)
     {
         return GameActions::Result(GameActions::Status::InvalidParameters, STR_CANT_INCREASE_MAP_SIZE_ANY_FURTHER, STR_NONE);
     }
-    if (_targetSize < 16)
+    if (_targetSize.x < 16 || _targetSize.y < 16)
```

Comment:
```suggestion
    if (_targetSize.x < MINIMUM_MAP_SIZE_TECHNICAL || _targetSize.y < MINIMUM_MAP_SIZE_TECHNICAL)
```
(This was off-by-one before.)